lib/pc_patch.c

```diff
@@ -544,3 +544,60 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+/** set schema for patch */
+PCPATCH*
+pc_patch_set_schema(const PCPATCH *patch, const PCSCHEMA *new_schema)
```

Comment:
Do you really want to expose the input patch as const, whereas its pcid is modified in the fast path?